<?xml version="1.0"?>

<launch>
    <!-- Launch pub_tf_posestamped for wamv2 -->
    <include file="$(find localization)/launch/pub_tf_posestamped.launch">
        <arg name="model_name" value="wamv2"/>
        <arg name="pub_tf" value="false"/>
    </include>

    <!-- Launch pub_tf_posestamped for wamv3 -->
    <include file="$(find localization)/launch/pub_tf_posestamped.launch">
        <arg name="model_name" value="wamv3"/>
    </include>

    <!-- Launch pub_tf_posestamped for wamv4 -->
    <include file="$(find localization)/launch/pub_tf_posestamped.launch">
        <arg name="model_name" value="wamv4"/>
    </include>

    <!-- Launch lidar_crop for wamv2 -->
    <include file="$(find lidar_crop)/launch/lidar_crop.launch">
        <arg name="veh" value="wamv2"/>
    </include>

    <!-- Launch lidar_crop for wamv3 -->
    <include file="$(find lidar_crop)/launch/lidar_crop.launch">
        <arg name="veh" value="wamv3"/>
    </include>

    <!-- Launch lidar_crop for wamv4 -->
    <include file="$(find lidar_crop)/launch/lidar_crop.launch">
        <arg name="veh" value="wamv4"/>
    </include>

    <!-- Launch usv_joydrive for wamv2 -->
    <include file="$(find vrx_gazebo)/launch/usv_joydrive.launch">
        <arg name="veh" value="wamv2"/>
        <arg name="use_namespace" value="true"/>
        <arg name="use_joy" value="false"/>
    </include>

    <!-- Launch usv_joydrive for wamv3 -->
    <include file="$(find vrx_gazebo)/launch/usv_joydrive.launch">
        <arg name="veh" value="wamv3"/>
        <arg name="use_namespace" value="true"/>
        <arg name="use_joy" value="false"/>
    </include>

    <!-- Launch usv_joydrive for wamv4 -->
    <include file="$(find vrx_gazebo)/launch/usv_joydrive.launch">
        <arg name="veh" value="wamv4"/>
        <arg name="use_namespace" value="true"/>
        <arg name="use_joy" value="false"/>
    </include>
</launch>
